### PR TITLE
[terra-action-header] Update docs to use MDX syntax

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
Updated action-header docs to use MDX syntax.

### CODE REVIEW

**Comment by pranav300** [comment]
This [Link](https://github.com/cerner/terra-core/pull/2802/files#diff-effa7949bad62054ed400858b5cd6345R41) will be broken until the [PR for base](https://github.com/cerner/terra-core/pull/2803) is merged.

File: packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx [hunk 1]
```diff
@@ -25,7 +38,7 @@ This component requires the following peer dependencies be installed in your app
 ## Implementation Notes:
 The Action-Header component must be composed inside the [Base][1] component with a locale in order for it to load the correct translation strings.
 
-[1]: https://github.com/cerner/terra-core/tree/master/packages/terra-base/docs
+[1]: https://github.com/cerner/terra-core/tree/master/packages/terra-base/src/terra-dev-site/doc/base/Base.1.doc.mdx
 
 ## Usage
```
Comment (StephenEsser) [inline packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx]: We should grab a few more opinions on this before making any changes, but I think we should consider removing the usage section all together or reducing it to include only the import during these mdx uplifts. The static code blocks can easily become outdated and the living code examples below provide a good example implementation and guarantee code integrity. 

This [PR](https://github.com/cerner/terra-core/pull/2781/files) for example reduced the usage block to include only the import.

Comment (emilyrohrbough) [reply packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx]: +1 to reducing to the import. The code blocks can show working code examples

File: packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+import BackActionHeader from '../example/BackActionHeader?dev-site-example';
+import BackCloseActionHeader from '../example/BackCloseActionHeader?dev-site-example';
+import CloseActionHeader from '../example/CloseActionHeader?dev-site-example';
+import MaximizeCloseActionHeader from '../example/MaximizeCloseActionHeader?dev-site-example';
+import MinimizeCustomButtonActionHeader from '../example/MinimizeCustomButtonActionHeader?dev-site-example';
+import BackPreviousNextCloseActionHeader from '../example/BackPreviousNextCloseActionHeader?dev-site-example';
+import LongWrappingTextHeader from '../example/LongWrappingTextHeader?dev-site-example';
+
+import ActionHeaderPropsTable from '../../../../src/ActionHeader?dev-site-props-table';
+
```
Comment (emilyrohrbough) [inline packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx]: We should be rendering the Badge at the top of these docs

File: packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx (hunk 1 above)
Comment (pranav300) [reply packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx]: Updated here https://github.com/cerner/terra-core/pull/2802/commits/5339e5c49685a2febe7d1fd150600dc7892892f5.

File: packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx (hunk 2 above)
Comment (pranav300) [reply packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx]: updated here https://github.com/cerner/terra-core/pull/2802/commits/5339e5c49685a2febe7d1fd150600dc7892892f5.

Comment (emilyrohrbough) [reply packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx]: @pranav300 lets add these to the Changelog and Upgrade guides as well so they read the same way as when they were rendered with Doc Template

Comment (pranav300) [reply packages/terra-action-header/src/terra-dev-site/doc/action-header/ActionHeader.1.doc.mdx]: updated here https://github.com/cerner/terra-core/pull/2802/commits/436ab51b2819072f156c3711708febc83c3bd277.

**Pull request merged**